tests/unit/functions.py

```diff
@@ -78,3 +84,28 @@ def it_fails_when_there_are_locks(self, tmpdir):
 
     def it_passes_when_there_are_no_locks(self, tmpdir):
         assert show_runaway_processes(tmpdir.strpath) is None
+
+
+class DescribeTerminateRunawayProcesses(object):
+
+    @pytest.yield_fixture
+    def mock_subprocess_call(self):
+        with mock.patch.object(pgctl.subprocess, 'call', autospec=True) as mock_call:
+            yield mock_call
+
+    def it_kills_processes_holding_the_lock(self, tmpdir, mock_subprocess_call):
+        lockfile = tmpdir.ensure('lock')
+        lock = lockfile.open()
+
```

Comment:
I'd instead set the lock fd as inheritable, Popen a 'sleep infinity' (which holds the lock forever), then assert that the sleep died after calling terminate-runaway-processes.